fluid/PaddleCV/int8/utility.py

```diff
@@ -0,0 +1,336 @@
+        num_merged_bins = i/num_quantized_bins
+        candidate_distr_Q_quantized = [0]*num_quantized_bins
+        j_start = 0
+        j_end = num_merged_bins
+        for idx in xrange(num_quantized_bins):
+	    candidate_distr_Q_quantized[idx] = sum(candidate_distr_Q[j_start:j_end])
+            j_start += num_merged_bins
+            j_end += num_merged_bins
+            if (idx+1) == num_quantized_bins - 1:
+	        j_end = i
+	candidate_distr_Q = expand_quantized_bins(candidate_distr_Q_quantized, reference_distr_bins)
+        Q_sum = sum(candidate_distr_Q)
+        kl_divergence = safe_entropy(reference_distr_P, P_sum, candidate_distr_Q, Q_sum)
+        if not kl_inited:
+            min_kl_divergence = kl_divergence
+            min_kl_index = i
+            kl_inited = True
+        elif kl_divergence < min_kl_divergence:
+	    min_kl_divergence = kl_divergence
+            min_kl_index = i
+	else:
+	    pass
+    if min_kl_index == 0:
+        while starting_iter > 0:
+            if hist[starting_iter] == 0:
+                starting_iter -= 1
+                continue
+            else:
+                break
+        min_kl_index = starting_iter
+    return (min_kl_index+0.5)*bin_width
+
+def analysis_program_var(program):
+    conv_op_index = [index for index, value in enumerate(program.global_block().ops) if value.type == 'conv2d']
+    weights_var_name = []
+    conv_input_var_name = []
+    conv_output_var_name = []
+    quantization_var_name = []
+    
+    for i in conv_op_index[1:]:
+        weights_var_name.append(program.current_block().ops[i].input('Filter')[0])
+        conv_input_var_name.append(program.current_block().ops[i].input('Input')[0])
+        conv_output_var_name.append(program.current_block().ops[i].output('Output')[0])
+    
+    quantize_dequantize_index = get_quantize_dequantize_combination(program)
+    
+    for i in range(0, len(quantize_dequantize_index), 2):
+        output_index = quantize_dequantize_index[0] - 1
+        input_index = quantize_dequantize_index[1] 
+        for j in range(quantize_dequantize_index[1] + 1, len(program.current_block().ops) - 1):
+            if program.current_block().ops[j].type in ("pool2d") and program.current_block().ops[j + 1].type not in ("pool2d"):
+                input_index = j
+                break
+            else:
+                continue
+        output_var_name = program.current_block().ops[output_index].output_names[0]
+        output_var = program.current_block().ops[output_index].output(output_var_name)[0]
+        input_var_name = program.current_block().ops[input_index].output_names[0]
+        input_var = program.current_block().ops[input_index].output(input_var_name)[0]
+        quantization_var_name.append(output_var)
+        quantization_var_name.append(input_var)
+    
+    return conv_input_var_name, conv_output_var_name, weights_var_name, quantization_var_name
+
+def check_op_type_by_input(program, var_name, start_index=0):
```

Comment:
why does it want to check? why is `conv2d` `pool2d` special? Maybe the function needs a better name or comment?

Reply:
Right, we need a better name. Basically, conv and pooling are two major primitives running under INT8.